jobs/integration/validation.py

```diff
@@ -1462,25 +1462,27 @@ def _find_relation(*specs):
             pytest.fail(f"Timed out waiting for {db_name} to go away")
 
 
-@pytest.mark.skip_if_apps(
-    # skip this test if ceph-mon and ceph-osd are already installed
-    lambda apps: all(a in apps for a in ["ceph-mon", "ceph-osd"])
-)
 @pytest.mark.usefixtures("ceph_apps")
-async def test_ceph(model, log_open):
-    log.info("waiting for csi to settle")
-    unit = model.applications["kubernetes-control-plane"].units[0]
-    await retry_async_with_timeout(
-        verify_ready,
-        (unit, "po", ["csi-rbdplugin", "csi-cephfsplugin"]),
-        timeout_msg="CSI pods not ready!",
-    )
-    # create pod that writes to a pv from ceph
-    kwds = {"provisioner": "csi-rbdplugin-provisioner", "debug_open": log_open}
-    await validate_storage_class(model, "ceph-xfs", "Ceph", **kwds)
-    await validate_storage_class(model, "ceph-ext4", "Ceph", **kwds)
-    kwds["provisioner"] = "cephfs-provisioner"
-    await validate_storage_class(model, "cephfs", "Ceph", **kwds)
+class TestCeph:
+    async def test_plugins_installed(self, model):
+        log.info("waiting for csi to settle")
+        unit = model.applications["kubernetes-control-plane"].units[0]
+        await retry_async_with_timeout(
+            verify_ready,
+            (unit, "po", ["csi-rbdplugin", "csi-cephfsplugin"]),
+            timeout_msg="CSI pods not ready!",
+        )
+
+    @pytest.mark.parametrize("storage_class", ["ceph-xfs", "ceph-ext4", "cephfs"])
+    async def test_storage_class(self, model, log_open, storage_class):
+        # create pod that writes to a pv from ceph
+        kwds = dict(debug_open=log_open)
+        if storage_class == "cephfs":
+            kwds["provisioner"] = "cephfs-provisioner"
+        else:
+            kwds["provisioner"] = "csi-rbdplugin-provisioner"
+
+        await validate_storage_class(model, storage_class, "Ceph", **kwds)
```

Comment:
split the previous `test_ceph` into what amounts to 4 tests -- one to see if the plugins are installed correctly, and 3 for each of the storage classes associated with these charms

the `ceph_apps` pytest fixture is changed from a function-scoped fixture that cleans up the apps on completion of the test function, to class-scoped which cleans up when all the tests are completed associated with the `TestCeph` class.


Finally -- move the check to skip the tests into the `ceph_apps` fixture rather than use `skip_if_apps` marker